src/io.rs

```diff
@@ -525,4 +570,31 @@ mod tests {
             "collision with foo at offset 50"
         );
     }
+
+    #[test]
+    fn limit_writer_test() {
+        let data: Vec<u8> = (0..100).collect();
+
+        // limit larger than data
+        let mut outbuf: Vec<u8> = Vec::new();
+        let mut lim = LimitWriter::new(&mut outbuf, 150, "foo".into());
+        lim.write_all(&data).unwrap();
+        lim.flush().unwrap();
+        assert_eq!(data, outbuf);
+
+        // limit exactly equal to data
+        let mut outbuf: Vec<u8> = Vec::new();
+        let mut lim = LimitWriter::new(&mut outbuf, 100, "foo".into());
+        lim.write_all(&data).unwrap();
+        lim.flush().unwrap();
+        assert_eq!(data, outbuf);
+
+        // limit smaller than data
+        let mut outbuf: Vec<u8> = Vec::new();
+        let mut lim = LimitWriter::new(&mut outbuf, 90, "foo".into());
+        assert_eq!(
+            lim.write_all(&data).unwrap_err().to_string(),
+            "collision with foo at offset 90"
+        );
+    }
```

Comment:
There's some marginal value in testing multiple write calls per writer, particularly:
- checking that a write overrun succeeds up to the limit and then fails when `write` is called again at the limit,
- checking that a zero-length write succeeds at the limit.